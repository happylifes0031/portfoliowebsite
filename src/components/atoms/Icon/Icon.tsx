import React from "react";
// @ts-ignore
import styled from "@emotion/styled";
// @ts-ignore
import { css, SerializedStyles } from "@emotion/core";

interface StyledIconProps {
  active: boolean;
}

const StyledIcon = styled.span(
  (props: StyledIconProps): SerializedStyles => {
    return css([
      css`
        margin-right: 0.8em;
        margin-top: 0.8em;
        transition-property: all;
        transition-duration: 1s;
        display: flex;
        justify-content: center;
        align-items: center;
        -webkit-filter: grayscale(90%);
        -moz-filter: grayscale(90%);
        -o-filter: grayscale(90%);
        -ms-filter: grayscale(90%);
        filter: grayscale(90%);
      `,
      props.active &&
        css`
          display: block;
          height: inherit;
          border-radius: 5px;
          -webkit-filter: grayscale(0%);
          -moz-filter: grayscale(0%);
          -o-filter: grayscale(0%);
          -ms-filter: grayscale(0%);
          filter: grayscale(0%);
        `
    ]);
  }
);

const StyledImage = styled.img(
  (props: StyledIconProps): SerializedStyles => {
    return css([
      `
        width: 40px;
       height: 40px;
      transition-property: all;
      transition-duration: 1s;
    `,
      props.active &&
        `    
        width: 50px;
        height: 50px;
        border-radius:5px;
        
          -webkit-box-shadow: 1px 1px 12px 0px rgba(0,0,0,0.35);
            -moz-box-shadow: 1px 1px 12px 0px rgba(0,0,0,0.35);
            box-shadow: 1px 1px 12px 0px rgba(0,0,0,0.35);
    `
    ]);
  }
);

const Icon = ({ element, active }) => (
  <StyledIcon className="icon" id={element.icon} active={active}>
    <StyledImage
      src={"./images/icons/" + element.icon + ".png"}
      alt={element.icon + " icon"}
      active={active}
    />
  </StyledIcon>
);

export default Icon;
