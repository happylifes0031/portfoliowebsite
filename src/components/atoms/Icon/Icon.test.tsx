import React from "react";
import renderer from "react-test-renderer";
import { matchers } from "jest-emotion";
import "jest-styled-components";

import Icon from "./Icon";
expect.extend(matchers);

describe("Icon", () => {
  const icon = renderer
      .create(<Icon active={true} element={{ icon: "blender", range: "12" }} />)
      .toJSON();

  it("Renders a snapshot", () => {
    expect(icon).toMatchSnapshot();
  });

  it("Icon has filter at 0%, when active", () => {
    expect(icon).toHaveStyleRule("filter", "grayscale(0%)");
    expect(icon).not.toHaveStyleRule("filter", "grayscale(90%)");
  });

  it("Scales image inside the icon", () => {
    const image = icon.children[0];
    expect(image).toHaveStyleRule("width", "50px");
    expect(image).not.toHaveStyleRule("width", "40px");
  });
});
