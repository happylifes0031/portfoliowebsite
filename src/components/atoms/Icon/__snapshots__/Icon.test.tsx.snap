// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Renders a snapshot 1`] = `
<span
  className="icon css-16j44wd"
  id="blender"
>
  <img
    alt="blender icon"
    className="css-14hsflx"
    src="./images/icons/blender.png"
  />
</span>
`;
